src/query/storages/fuse/src/operations/append.rs

```diff
@@ -66,7 +65,7 @@ impl FuseTable {
                         BlockCompactorForCopy::new(block_thresholds),
                     )?))
                 })?;
-                pipeline.try_resize(size)?;
+                pipeline.try_resize(ctx.get_settings().get_max_threads()? as usize)?;
```

Comment:
`ctx.get_settings()` has a certain overhead, it is recommended to only perform one operation.

Reply:
can only access to a  `dyn TableContext` here.

Reply:
maybe we can cache a settings in TableContext